### PR TITLE
Add NICO dataset with basic data-loading/downloading functionality (no biasing schemes yet).

### PR DESCRIPTION
**Pull request by MylesBartlett**

### CODE REVIEW

File: bolts/data/datasets/utils.py [hunk 1]
```diff
@@ -71,7 +73,7 @@ class BinarySample(NamedTuple):
 
 class TernarySample(NamedTuple):
     x: Tensor | np.ndarray | Image.Image
-    s: Tensor | float
+    z: Tensor | float
```
Comment (thomkeh) [inline bolts/data/datasets/utils.py:76]: ?

File: bolts/data/datasets/vision/nico.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        return metadata
+
+    def __len__(self) -> int:
+        return len(self.x)
+
+    def __getitem__(self, index: int) -> TernarySample:
+        image = load_image(self._base_dir / self.x[index], backend=self._il_backend)
+        image = apply_image_transform(image=image, transform=self.transform)
+        target = self.y[index]
+        return TernarySample(x=image, s=self.z[index], y=target)
```
Comment (thomkeh) [inline bolts/data/datasets/vision/nico.py:152]: It still says `s=` here.

File: bolts/data/datasets/utils.py (hunk 1 above)
Comment (MylesBartlett) [reply bolts/data/datasets/utils.py:76]: ?

Comment (thomkeh) [reply bolts/data/datasets/utils.py:76]: Why would you give s such a confusing name?

Comment (MylesBartlett) [reply bolts/data/datasets/utils.py:76]: For the very reason that it's not s - it's more general than that. z is hardly an arbitrary choice to go with x and y and indeed it's fairly common in the literature to use it to denote some side information.

Comment (olliethomas) [reply bolts/data/datasets/utils.py:76]: this will cause problems for `bolts.fair.models` which assume `batch.x`, `batch.s` and `batch.y`

**Pull request merged**